libraries/botbuilder-dialogs/src/componentDialog.ts

```diff
@@ -243,10 +240,27 @@ export class ComponentDialog<O extends object = {}> extends DialogContainer<O> {
         return outerDC.endDialog(result);
     }
 
-    private createInnerDC(context: TurnContext, instance: DialogInstance) {
-        const dialogState = instance.state[PERSISTED_DIALOG_STATE] || { dialogStack: [] };
+    private createInnerDC(context: DialogContext, instance: DialogInstance);
+    private createInnerDC(context: TurnContext, instance: DialogInstance);
+    private createInnerDC(context: TurnContext | DialogContext, instance: DialogInstance) {
+
+        let dialogState = {} as DialogState;
+
+        if (!instance) {
+            instance = { state: {} } as DialogInstance;
+        }
+
+        dialogState = instance.state[PERSISTED_DIALOG_STATE] || { dialogStack: [] };
         instance.state[PERSISTED_DIALOG_STATE] = dialogState
-        const innerDC: DialogContext = new DialogContext(this.dialogs, context, dialogState);
+
+        let innerDC: DialogContext;
+
+        if(context instanceof DialogContext){
+            innerDC = new DialogContext(this.dialogs, context, dialogState);
+
+        }else{
+            innerDC = new DialogContext(this.dialogs, context, dialogState);
+        }
```

Comment:
This if-check is unnecessary due to differences in language between C# and TypeScript.

returning a `new DialogContext(this.dialogs, context, dialogState);` should suffice.